# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/sandbox/chromium/base/at_exit.h
# Commit: 370c40e49230
# Full Hash: 370c40e4923073a87845b57c1f4851bacd9663da
# Author: Bob Owen <bobowencode@gmail.com>
# Date: 2019-06-12 21:42:35
# Regressor Bug: 1552160
# File Overlap Count: 2
# Description:
#   Bug 1552160 Part 1: Roll-up of chromium sandbox update and mozilla patches to get a running browser. r=jld,aklotz,tjr,bobowen
#   
#   This updates security/sandbox/chromium/ files to chromium commit 84108231f6e6e0772fb9a4643679ce76aa771e67.
#   
#   Existing and new patches applied from security/sandbox/chromium-shim/patches/with_update/ to give a compiling and mostly working browser.
# ==============================================================================

diff -r bd5ecd72f76c -r 370c40e49230 security/sandbox/chromium/base/at_exit.h
--- a/security/sandbox/chromium/base/at_exit.h	Wed Jun 12 21:55:59 2019 +1200
+++ b/security/sandbox/chromium/base/at_exit.h	Wed Jun 12 11:10:48 2019 +0100
@@ -5,12 +5,12 @@
 #ifndef BASE_AT_EXIT_H_
 #define BASE_AT_EXIT_H_
 
-#include <stack>
-
 #include "base/base_export.h"
 #include "base/callback.h"
+#include "base/containers/stack.h"
 #include "base/macros.h"
 #include "base/synchronization/lock.h"
+#include "base/thread_annotations.h"
 
 namespace base {
 
@@ -62,9 +62,15 @@
 
  private:
   base::Lock lock_;
-  std::stack<base::Closure> stack_;
-  bool processing_callbacks_;
-  AtExitManager* next_manager_;  // Stack of managers to allow shadowing.
+
+  base::stack<base::Closure> stack_ GUARDED_BY(lock_);
+
+#if DCHECK_IS_ON()
+  bool processing_callbacks_ GUARDED_BY(lock_) = false;
+#endif
+
+  // Stack of managers to allow shadowing.
+  AtExitManager* const next_manager_;
 
   DISALLOW_COPY_AND_ASSIGN(AtExitManager);
 };